### PR TITLE
add reflection on health check

### PR DESCRIPTION
**Pull request by thatInfrastructureGuy**
Signed-off-by: Ashish Kulkarni <thatInfrastructureGuy@gmail.com>

### CODE REVIEW

**Comment by thatInfrastructureGuy** [comment]
Idea is to pull your package and have a ready-to-deploy infrastructure with health checks and reflection. Perhaps its a dump idea.

File: utils/grpc_health_check_provider.go [hunk 1]
```diff
@@ -33,4 +34,5 @@ func (s *GrpcHealthServer) Watch(*he.HealthCheckRequest, he.Health_WatchServer)
 
 func RegisterHealthCheckService(s *grpc.Server) {
 	he.RegisterHealthServer(s, new(GrpcHealthServer))
+	reflection.Register(s)
```
Comment (taiyangc) [inline utils/grpc_health_check_provider.go]: @JustinGaoF can you check if this is reasonable? I have only seen this line happening in main.go, not a helper.

Maybe @uchetron can chime in as well

Comment (JustinGaoF) [reply utils/grpc_health_check_provider.go]: @thatInfrastructureGuy  Can you explain the reason to add this line? I am not understand the reason, based on my test, the L36 is enough for provide healthcheck of gRPC. 

cc @taiyangc  @uchetron  FYI, need your idea too

Comment (taiyangc) [reply utils/grpc_health_check_provider.go]: From what I understand and what we use currently, reflection is an optional extension for easy discovery of services. https://github.com/grpc/grpc-go/blob/master/Documentation/server-reflection-tutorial.md

It should be done at the caller level IMO. But I intend to talk about this within the team on Thursday.

Comment (thatInfrastructureGuy) [reply utils/grpc_health_check_provider.go]: I understand. I saw the possibility of dup code for status-server and escrow:

```
utils.RegisterHealthCheckService(gRPCServer)
reflection.Register(gRPCServer)
```
Hence, the idea was to put reflection in `utils` package. But this is definitely optional and hence, I will close this PR.

**Pull request closed without merging**